### DISCUSSION TITLE
L11 cast function visibility

### DISCUSSION THREAD

**Discussion by Pronesti** [larastan/larastan#2087] [General]
Laravel documentation sets the new cast function to be protected. [link](https://laravel.com/docs/11.x/eloquent-mutators#attribute-casting)
But I had to set it public in order for Larastan to understand these new casts.

I would love some clarification on this new change.

**Comment by szepeviktor** [comment]
There could be a bug 🐞

https://github.com/larastan/larastan/blob/9e7233d88f4f10796fadb8a2d85c5f2b55277c76/UPGRADE.md?plain=1#L11

**Reply by Pronesti** [reply]
I'm not familiar enough with the framework's internals to assess the impact of this visibility change. However, I believe Larastan should support the protected visibility, which is a standard practice in the framework. Should I open an issue?

**Reply by calebdw** [reply]
This is a result of using the `OutOfClassScope`

https://github.com/larastan/larastan/blob/9e7233d88f4f10796fadb8a2d85c5f2b55277c76/src/Properties/ModelCastHelper.php#L251-L254
